lib/OpenQA/WebAPI/Controller/Step.pm

```diff
@@ -19,7 +31,7 @@ sub _init {
     my ($self) = @_;
 
     return 0 unless my $job = $self->app->schema->resultset('Jobs')->find($self->param('testid'));
-    my $module = $job->modules->search({name => $self->param('moduleid')})->first;
+    my $module = $job->modules->find({name => $self->param('moduleid')});
```

Comment:
There's no unique constraint on the name column alone so this query might return multiple results. Better limit it to one row:

```suggestion
    my $module = $job->modules->find({name => $self->param('moduleid')}, {rows => 1});
```

It is a bit unfortunate that the database schema's unique constraints don't match the way API parameters are specified but I suppose improving that is out of scope for this PR. Just ensure we don't end up with Perl warnings in the logs due to this limitation.

---

Please also check whether other changes of that kind fall into the same category. A query via `find` must not return more than one row.

Reply:
I moved all "find" related changes to https://github.com/os-autoinst/openQA/pull/4202